kingpin/actors/aws/cloudformation.py

```diff
@@ -0,0 +1,434 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2014 Nextdoor.com, Inc
+
+"""AWS.CloudFormation Actors"""
+
+import logging
+
+from boto import cloudformation
+from boto.exception import BotoServerError
+from concurrent import futures
+from retrying import retry
+from tornado import concurrent
+from tornado import gen
+from tornado import ioloop
+
+from kingpin import utils
+from kingpin.actors import base
+from kingpin.actors import exceptions
+from kingpin.actors.aws import settings as aws_settings
+from kingpin.constants import REQUIRED
+
+log = logging.getLogger(__name__)
+
+__author__ = 'Matt Wise <matt@nextdoor.com>'
+
+
+# This executor is used by the tornado.concurrent.run_on_executor()
+# decorator. We would like this to be a class variable so its shared
+# across RightScale objects, but we see testing IO errors when we
+# do this.
+EXECUTOR = futures.ThreadPoolExecutor(10)
+
+
+# Used by the retrying.retry decorator
+def retry_if_transient_error(exception):
+    return isinstance(exception, BotoServerError)
+
+
+class CloudFormationError(exceptions.RecoverableActorFailure):
+
+    """Raised on any generic CloudFormation error."""
+
+
+class InvalidTemplate(exceptions.UnrecoverableActorFailure):
+
+    """An invalid CloudFormation template was supplied."""
+
+
+class StackAlreadyExists(exceptions.RecoverableActorFailure):
```

Comment:
This and NotFound should probably inherit from CloudFormationError, just to respect the hierarchy